src/Management/src/Endpoint/HostBuilderWrapperExtensions.cs

```diff
@@ -135,7 +122,12 @@ public static void AddServicesActuator(this HostBuilderWrapper wrapper)
 
     public static void AddCloudFoundryActuator(this HostBuilderWrapper wrapper)
     {
-        wrapper.ConfigureServices(services => services.AddCloudFoundryActuator());
+        wrapper.ConfigureServices(services =>
+        {
+            services.AddCloudFoundryActuator();
+            services.AddCloudFoundrySecurity();
```

Comment:
I'm a little confused on why AddCloudFoundrySecurity is required as a separate call. My recollection was that the security pieces were automatically added with the CloudFoundry Actuator, but based on the changes here it seems like it wasn't all setup correctly before? I wonder if [this ](https://github.com/SteeltoeOSS/Steeltoe/issues/1293) is related

Reply:
What I did here is similar to Eureka and ConfigServer: use HttpClientFactory. The original code maintained a lazily-created long-lived `HttpClient` instance, which I replaced with pulling one from the factory each time. But the factory needs to be registered in the IoC container at startup. Since there already was `UseCloudFoundrySecurity` to activate the middleware, I introduced `AddCloudFoundrySecurity` to run these registrations. Because it's a new extension method and a prerequisite for `UseCloudFoundrySecurity`, I added a check to throw when `AddCloudFoundrySecurity` hasn't been called.

I don't sufficiently understand the full feature at play here to be able to test the end-to-end scenario, or determine whether a new extension is justified over calling it from an existing place. But what I do understand is there's an outgoing HTTP request where certificate validation can be turned off through configuration. So I just refactored that. I made `SecurityUtils` injectable, so it responds to configuration changes. The changes in this commit may accidentally fix existing bugs, I have insufficient background knowledge to determine that.

Reply:
The proposal is to move the logic inside `AddCloudFoundrySecurity` into `AddCloudFoundryActuator` and not have this separate extension method.